backend/test/resolver.spec.js

```diff
@@ -0,0 +1,91 @@
+const { gql } = require('apollo-server-express');
+const { createTestClient } = require('apollo-server-testing');
+const { getTestApolloServer } = require('../server.js');
+const jwt = require('jsonwebtoken');
+const server = getTestApolloServer();
+const initClient = createTestClient(server);
+const query = initClient.query;
+const mutate = initClient.mutate;
+
+// could not outsource these queries/mutations...
+const ALL_ITEMS_QUERY = gql`
+    query itemsQuery {
+        items {
+            id,
+            text,
+            editMode
+        }
+    }
+`;
+
+const CREATE_ENTRY = gql`
+    mutation createEntry($text: String!) {
+        createEntry(text: $text) {
+            id,
+            text,
+            editMode
+        }
+    }
+`;
+
+const DELETE_ENTRY = gql`
+    mutation DeleteEntry($id: String!) {
+        deleteEntry(id: $id)
+    }
+`;
+
+const LOGIN = gql`
+    mutation login($email: String!, $password:String!) {
+        login(email: $email, password: $password)
+    }
+`;
+
+describe('User is logged in', () => {
+    describe('Queries', () => {
+        it("has start todo items", () => {
+            query({ query: ALL_ITEMS_QUERY }).then((data) => {
+                expect(data.data.items).toHaveLength(4);
+                expect(data.data.items).toMatchObject([
+                    { id: 1 },
+                    { id: 2 },
+                    { id: 3 },
+                    { id: 4 }
+                ]);
+            });
+        });
+    });
+
+    describe('Mutations', () => {
+        const exampleText = "example text";
+        it("creates entry", () => {
+            mutate({ mutation: CREATE_ENTRY, variables: { text: exampleText } })
+                .then((data) => {
+                    expect(data.data.createEntry.id).toBe(123987123981723)
+                    expect(data.data.createEntry.id).toBeGreaterThanOrEqual(0)
+                    expect(typeof data.data.createEntry.id).toBe('string')
```

Comment:
How is that possible, does `toBe` not take into account the type of the subject? I mean how can a string `toBe` the number `123987123981723`?

Anyways, I think if you can expect the value directly, that's enough. If you want to add documentation for the reader, put that documentation into `describe` and `it` descriptions.